gen/top_gen.ml

```diff
@@ -401,6 +401,16 @@ let max_set = IntSet.max_elt
               i,c::cs,f@fs
         with NoObserver -> build_observers p i x vss
 
+  let filter_map f =
+    let rec aux accu = function
+      | [] -> List.rev accu
+      | x :: l ->
+          match f x with
+          | None -> aux accu l
+          | Some v -> aux (v :: accu) l
+    in
+    aux []
+
```

Comment:
It is fine to have `filter_map` coded in place of calling the standard library `List.filter_map` as it preserved compatibility starting from OCaml 4.05. However I'd rather see this function in the `lib/misc.ml` file.

Reply:
Thanks for the direction here. I knew that this wasn't the place for this, but didn't know any better as to where else to put it :)